source/containers/Form/hooks/formActions.ts

```diff
@@ -367,9 +368,17 @@ export function validateAllStepAnswers( state: FormReducerState, onErrorCallback
 
     if (itemsToValidate.length > 0) {
       itemsToValidate.forEach(validationItem => {
+        let shouldValidate = true;
         const answer = state.formAnswers[validationItem.id] || '';
         dirtyFields[validationItem.id] = true;
-        state = validateAnswer(state, { [validationItem.id]: answer }, validationItem.id);
+
+        if(validationItem.hasCondition && validationItem.conditionalOn) {
+          shouldValidate = evaluateConditionalExpression(validationItem.conditionalOn, state.formAnswers, state.allQuestions);
```

Comment:
A thought here would be to move the logic for determine if a validation should run or not in a separate function. For me it's a bit confusing i terms of readability with the shouldValidate variabel. I do get the purpose, but i would rather have one line that looks something like:

```JS
if (shouldValidateAnswer(validationItem)) {
   state = validateAnswer(state, { [validationItem.id]: answer }, validationItem.id)
}
```

Reply:
Another thought is to move this logic to the validateAnswer function, and thereby keeping it in that scope :)

Reply:
Completly out of this scope, but the passing of arguments here feels very redundant.
```JS 
validateAnswer(state, { [validationItem.id]: answer }, validationItem.id)`
```

Would be way better to just pass the values and the construct the object within the function itself.
```JS 
validateAnswer(state, validationItem.id, answer)
```

So feel free to refactor this if it's a simple fix (Y)

Reply:
> A thought here would be to move the logic for determine if a validation should run or not in a separate function. For me it's a bit confusing i terms of readability with the shouldValidate variabel. I do get the purpose, but i would rather have one line that looks something like:
> 
> ```js
> if (shouldValidateAnswer(validationItem, formAnswers, allQuestions)) {
>    state = validateAnswer(state, { [validationItem.id]: answer }, validationItem.id)
> }
> ```

I like this answer more than moving the logic into `validateAnswer` itself 🙂. IMO functions should only do what they're saying they do, and skipping validation is not the same as having a valid answer, otherwise the function should be something like `maybeValidateAnswer` and that is even more confusing 😅.

Reply:
> Completly out of this scope, but the passing of arguments here feels very redundant.
> 
> ```js
> validateAnswer(state, { [validationItem.id]: answer }, validationItem.id)`
> ```
> 
> Would be way better to just pass the values and the construct the object within the function itself.
> 
> ```js
> validateAnswer(state, validationItem.id, answer)
> ```
> 
> So feel free to refactor this if it's a simple fix (Y)

Looked into this and it is not a simple fix; `validateAnswer` is used by the state reducer system where the structure of the payload is tailored around how the `validateAnswer` signature looks right now.